tests/test_orquestra_device.py

```diff
@@ -291,3 +421,418 @@ def circuit():
+            )
+
+            # Disable random uuid generation
+            m.setattr(uuid, "uuid4", lambda *args: test_uuid)
+
+            res = dev.batch_execute(circuits)
+
+            # Correct order of results is expected
+            assert np.allclose(res[0], test_batch_res0)
+            assert np.allclose(res[1], test_batch_res1)
+            assert np.allclose(res[2], test_batch_res2)
+            file0_kept = os.path.exists(tmpdir.join(f"expval-{test_uuid}-0.yaml"))
+            file1_kept = os.path.exists(tmpdir.join(f"expval-{test_uuid}-1.yaml"))
+            file2_kept = os.path.exists(tmpdir.join(f"expval-{test_uuid}-2.yaml"))
+
+        # Check that workflow files were either all kept or all deleted
+        files_kept = file0_kept and file1_kept and file2_kept
+        assert files_kept and file0_kept if keep else not files_kept
+
+        qml.disable_tape()
+
+    @pytest.mark.parametrize("dev", ["orquestra.forest", "orquestra.qiskit", "orquestra.qulacs"])
+    def test_identity_multiple_tape(self, dev, tmpdir, monkeypatch):
+        """Test computing the expectation value of the identity for multiple
+        return values."""
+        qml.enable_tape()
+
+        dev = qml.device(dev, wires=2, keep_files=False)
+
+        with qml.tape.QuantumTape() as tape1:
+            qml.RX(0.133, wires=0)
+            qml.expval(qml.Identity(wires=[0]))
+
+        with qml.tape.QuantumTape() as tape2:
+            qml.RX(0.432, wires=0)
+            qml.expval(qml.Identity(wires=[0]))
+            qml.expval(qml.Identity(wires=[1]))
+
+        circuits = [tape1, tape2]
+
+        test_uuid = "1234"
+        with monkeypatch.context() as m:
+            m.setattr(pennylane_orquestra.cli_actions, "user_data_dir", lambda *args: tmpdir)
+
+            # Disable submitting to Orquestra by mocking Popen
+            m.setattr(subprocess, "Popen", lambda *args, **kwargs: MockPopen())
+            m.setattr(
+                pennylane_orquestra.orquestra_device,
+                "loop_until_finished",
+                lambda *args, **kwargs: None,
+            )
+
+            # Disable random uuid generation
+            m.setattr(uuid, "uuid4", lambda *args: test_uuid)
+
+            res = dev.batch_execute(circuits)
+
+            # No workflow files were created as only computed with identities
+            assert not os.path.exists(tmpdir.join(f"expval-{test_uuid}.yaml"))
+            assert not os.path.exists(tmpdir.join(f"expval-{test_uuid}.yaml"))
+
+            expected = [np.ones(1), np.ones(2)]
+
+            for r, e in zip(res, expected):
+                assert np.allclose(r, e)
+
+        qml.disable_tape()
+
+    @pytest.mark.parametrize("mock_res", [[], "res", {"a": 3}, {"a": []}])
+    def test_wrong_result_format_multiple_steps(self, mock_res, monkeypatch):
+        """Test that the resource details defined when the device was created
+        are passed to generate the workflow."""
```

Comment:
Does this docstring match the test?

Reply:
Ooops, no, good catch! Thank you, updated